tensorflow/compiler/mlir/tosa/transforms/legalize_tf.cc

```diff
@@ -2003,7 +2019,7 @@ LogicalResult ConvertTFFakeQuantWithMinMaxVarsOp::matchAndRewrite(
     Operation* op, PatternRewriter& rewriter) const {
   auto tf_fakequant_op = cast<TF::FakeQuantWithMinMaxVarsOp>(op);
 
-  auto output_type =
+  RankedTensorType output_type =
       tf_fakequant_op.getResult().getType().dyn_cast<RankedTensorType>();
```

Comment:
For future reference: `auto` is perfectly fine when the type is obvious from the context, like when it appears in the "cast" template parameter on the RHS like here.

In general the guideline is to use `auto` when it makes the code more readable though (like iterator and other hard to write types).